lib/Doctrine/DBAL/Connection.php

```diff
@@ -96,6 +96,13 @@ class Connection implements DriverConnection
     protected $_conn;
 
     /**
+     * The connection name.
+     *
+     * @var string
+     */
+    protected $_name;
```

Comment:
This does not make any sense. Naming connections is only needed when you have multiple connections, but the connection has strictly no reason to be aware of it.
The implementation looking for the shard manager should be changed in the bundle to avoid this meaningless change in DBAL